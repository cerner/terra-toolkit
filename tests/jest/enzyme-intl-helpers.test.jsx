import React from 'react';
import { FormattedMessage, injectIntl } from 'react-intl';
import {
  shallowWithIntl,
  mountWithIntl,
  renderWithIntl,
  mockIntl,
} from '../../src/enzyme-intl';

const FormattedMessageSubject = props => (
  <div>
    <FormattedMessage id="TerraToolkit.helloWorld" />
    <FormattedMessage id="TerraToolkit.buttonText">
      {text => <button type="button">{text}</button>}
    </FormattedMessage>
    {JSON.stringify(props, undefined, 2)}
  </div>
);

const InjectIntlSubject = injectIntl(({ intl, ...props }) => (
  <div>
    {intl.formatMessage({ id: 'TerraToolkit.helloWorld' })}
    <button type="button">{intl.formatMessage({ id: 'TerraToolkit.buttonText' })}</button>
    {JSON.stringify(props, undefined, 2)}
  </div>
));

describe('shallowWithIntl', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = shallowWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = shallowWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('mountWithIntl', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = mountWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = mountWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('renderWithIntl', () => {
  describe('using FormattedMessage', () => {
    const subject = <FormattedMessageSubject foo="bar" />;
    const wrapper = renderWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('using injectIntl', () => {
    const subject = <InjectIntlSubject foo="bar" />;
    const wrapper = renderWithIntl(subject);

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});

describe('mockIntl', () => {
  const subject = intl => intl.formatMessage({ id: 'TerraToolkit.foo' });
  const expected = 'TerraToolkit.foo';

  describe('when used in a method expecting the react-intl intl object', () => {
    it('should match the expected output', () => {
      expect(subject(mockIntl)).toEqual(expected);
    });
  });
});
