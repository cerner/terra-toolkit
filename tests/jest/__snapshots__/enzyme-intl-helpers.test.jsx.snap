// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mountWithIntl using FormattedMessage should match the snapshot 1`] = `
<FormattedMessageSubject
  foo="bar"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <div>
    <FormattedMessage
      id="TerraToolkit.helloWorld"
      values={Object {}}
    >
      <span>
        TerraToolkit.helloWorld
      </span>
    </FormattedMessage>
    <FormattedMessage
      id="TerraToolkit.buttonText"
      values={Object {}}
    >
      <button
        type="button"
      >
        TerraToolkit.buttonText
      </button>
    </FormattedMessage>
    {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
  </div>
</FormattedMessageSubject>
`;

exports[`mountWithIntl using injectIntl should match the snapshot 1`] = `
<InjectIntl(Component)
  foo="bar"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
>
  <Component
    foo="bar"
    intl={
      Object {
        "defaultFormats": Object {},
        "defaultLocale": "en",
        "formatDate": [Function],
        "formatHTMLMessage": [Function],
        "formatMessage": [Function],
        "formatNumber": [Function],
        "formatPlural": [Function],
        "formatRelative": [Function],
        "formatTime": [Function],
        "formats": Object {},
        "formatters": Object {
          "getDateTimeFormat": [Function],
          "getMessageFormat": [Function],
          "getNumberFormat": [Function],
          "getPluralFormat": [Function],
          "getRelativeFormat": [Function],
        },
        "locale": "en",
        "messages": null,
        "now": [Function],
        "onError": [Function],
        "textComponent": "span",
        "timeZone": null,
      }
    }
  >
    <div>
      TerraToolkit.helloWorld
      <button
        type="button"
      >
        TerraToolkit.buttonText
      </button>
      {
  "foo": "bar"
}
    </div>
  </Component>
</InjectIntl(Component)>
`;

exports[`renderWithIntl using FormattedMessage should match the snapshot 1`] = `
<div>
  <span>
    TerraToolkit.helloWorld
  </span>
  <button
    type="button"
  >
    TerraToolkit.buttonText
  </button>
  {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
</div>
`;

exports[`renderWithIntl using injectIntl should match the snapshot 1`] = `
<div>
  TerraToolkit.helloWorld
  <button
    type="button"
  >
    TerraToolkit.buttonText
  </button>
  {
  "foo": "bar"
}
</div>
`;

exports[`shallowWithIntl using FormattedMessage should match the snapshot 1`] = `
<div>
  <FormattedMessage
    id="TerraToolkit.helloWorld"
    values={Object {}}
  />
  <FormattedMessage
    id="TerraToolkit.buttonText"
    values={Object {}}
  >
    <Component />
  </FormattedMessage>
  {
  "foo": "bar",
  "intl": {
    "locale": "en",
    "messages": {},
    "formats": {},
    "timeZone": null,
    "textComponent": "span",
    "defaultLocale": "en",
    "defaultFormats": {},
    "formatters": {}
  }
}
</div>
`;

exports[`shallowWithIntl using injectIntl should match the snapshot 1`] = `
<Component
  foo="bar"
  intl={
    Object {
      "defaultFormats": Object {},
      "defaultLocale": "en",
      "formatDate": [Function],
      "formatHTMLMessage": [Function],
      "formatMessage": [Function],
      "formatNumber": [Function],
      "formatPlural": [Function],
      "formatRelative": [Function],
      "formatTime": [Function],
      "formats": Object {},
      "formatters": Object {
        "getDateTimeFormat": [Function],
        "getMessageFormat": [Function],
        "getNumberFormat": [Function],
        "getPluralFormat": [Function],
        "getRelativeFormat": [Function],
      },
      "locale": "en",
      "messages": null,
      "now": [Function],
      "onError": [Function],
      "textComponent": "span",
      "timeZone": null,
    }
  }
/>
`;
